include/libtorrent/bencode.hpp

```diff
@@ -86,9 +86,12 @@ namespace libtorrent
 
 	namespace detail
 	{
-		template <class OutIt>
-		int write_integer(OutIt& out, entry::integer_type val)
+		template <class OutIt, class In, typename Cond
+			= typename std::enable_if<std::is_integral<In>::value>::type>
+		int write_integer(OutIt& out, In data)
 		{
+			entry::integer_type val = entry::integer_type(data);
```

Comment:
you should make this ``const``, and actually, there's a shorter form:

``entry::integer_type const val(data);``